src/test/java/com/wikia/webdriver/common/core/drivers/Browsers/FirefoxBrowser.java

```diff
@@ -0,0 +1,84 @@
+package com.wikia.webdriver.common.core.drivers.Browsers;
+
+import com.wikia.webdriver.common.core.ExtHelper;
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.drivers.BrowserAbstract;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxProfile;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+
+public class FirefoxBrowser extends BrowserAbstract {
+
+  private static FirefoxProfile firefoxProfile;
+
+  @Override
+  public WikiaWebDriver setInstance(){
+    // Windows 8 requires to set webdriver.firefox.bin system variable
+    // to path where executive file of FF is placed
+    if ("WINDOWS 8".equalsIgnoreCase(System.getProperty("os.name"))) {
+      System.setProperty("webdriver.firefox.bin", "c:" + File.separator + "Program Files (x86)"
+                                                  + File.separator + "Mozilla Firefox"
+                                                  + File.separator + "Firefox.exe");
+    }
+
+    // Check if user who is running tests have write access in ~/.mozilla dir and home dir
+    if ("LINUX".equals(System.getProperty("os.name").toUpperCase())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1157)